OpenRA.Game/Map/Map.cs

```diff
@@ -695,6 +785,34 @@ public CPos CellContaining(WPos pos)
 			return new CPos(u, v);
 		}
 
+		public PPos ProjectedCellCovering(WPos pos)
+		{
+			var projectedPos = pos - new WVec(0, pos.Z, pos.Z);
+			return (PPos)CellContaining(projectedPos).ToMPos(TileShape);
+		}
+
+		static readonly PPos[] NoProjectedCells = { };
+		public PPos[] ProjectedCellsCovering(MPos uv)
+		{
+			// Shortcut for mods that don't use heightmaps
+			if (MaximumTerrainHeight == 0)
+				return new PPos[] { (PPos)uv };
+
+			if (!cellProjection.Contains(uv))
+				return NoProjectedCells;
+
+			return cellProjection[uv];
+		}
+
+		public MPos[] Unproject(PPos puv)
```

Comment:
Yeah, this will be used in future PRs that resolve the `// TODO: Account for terrain height` through Map.